hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NodeEndpoint.java

```diff
@@ -171,4 +184,144 @@ private DatanodeStorageReport getStorageReport(DatanodeDetails datanode) {
     long committed = nodeStat.getCommitted().get();
     return new DatanodeStorageReport(capacity, used, remaining, committed);
   }
+
+  /**
+   * Removes datanodes from Recon's memory and nodes table in Recon DB.
+   * @param uuids the list of datanode uuid's
+   *
+   * @return JSON response with failed, not found and successfully removed datanodes list.
+   */
+  @PUT
+  @Path("/remove")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response removeDatanodes(List<String> uuids) {
+    List<DatanodeMetadata> failedDatanodes = new ArrayList<>();
+    List<DatanodeMetadata> notFoundDatanodes = new ArrayList<>();
+    List<DatanodeMetadata> removedDatanodes = new ArrayList<>();
+    Map<String, String> failedNodeErrorResponseMap = new HashMap<>();
+
+    Preconditions.checkNotNull(uuids, "Datanode list argument should not be null");
+    Preconditions.checkArgument(!uuids.isEmpty(), "Datanode list argument should not be empty");
+    try {
+      for (String uuid : uuids) {
+        DatanodeDetails nodeByUuid = nodeManager.getNodeByUuid(uuid);
+        try {
+          if (preChecksSuccess(nodeByUuid, failedNodeErrorResponseMap)) {
+            removedDatanodes.add(DatanodeMetadata.newBuilder()
+                .withHostname(nodeManager.getHostName(nodeByUuid))
+                .withUUid(uuid)
+                .withState(nodeManager.getNodeStatus(nodeByUuid).getHealth())
+                .build());
+            nodeManager.removeNode(nodeByUuid);
```

Comment:
Log could be added right after a successful operation that conclusively removes a node from the system.

```
    // Log successful removal
    LOG.info("Successfully removed node: {}", nodeByUuid.getUuidString());
```

Reply:
Done.